paimon-common/src/main/java/org/apache/paimon/predicate/PredicateBuilder.java

```diff
@@ -278,8 +279,20 @@ public static Object convertJavaObject(DataType literalType, Object o) {
                 int scale = decimalType.getScale();
                 return Decimal.fromBigDecimal((BigDecimal) o, precision, scale);
             case TIMESTAMP_WITHOUT_TIME_ZONE:
-            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
                 Timestamp timestamp;
+                if (o instanceof java.sql.Timestamp) {
+                    timestamp = Timestamp.fromSQLTimestamp((java.sql.Timestamp) o);
+                } else if (o instanceof Instant) {
+                    timestamp =
+                            Timestamp.fromInstant(
+                                    ((Instant) o).plusMillis(TimeUnit.HOURS.toMillis(8)));
```

Comment:
Don't hard code, please take a look to `DateTimeUtils`.

Reply:
OK